iogt/templates/base.html

```diff
@@ -41,9 +44,14 @@
             </div>
         </div>
 
-        {# Global javascript #}
-        <script type="text/javascript" src="{% static 'js/iogt.js' %}"></script>
-        {% block extra_js %}{% endblock %}
+         {# Global javascript #}    
 
+        <script>
+        function cache() {
```

Comment:
Is there a good reason why this function is inline rather than in a separate JS file?

I think it would be better to always register the service worker automatically, if possible, and activate caching as a separate step. We would only want to show the option to "Download website" if the service worker is successfully registered.

Reply:
Agreed. But since our sw.js is inside the template folder, I do not have access to JS functions that is inside sw.js. So I had to bring the function inline inside base.html.

Reply:
PS- The code here however will see further changes when addressing https://github.com/unicef/iogt/issues/117

Reply:
Alright, I think the inlining is not a big issue, so that can stay.

I would still want the service worker to be registered as soon as the page is loaded, regardless of whether caching is required or not. It is possible to [communicate with the worker via events](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage). So I suggest that the 'download' button, when clicked, sends a message to the worker, which then sets up the caching.

Reply:
Response from Kish via Slack:

> so I spent a few hours yesterday trying to make a fix as per your comment in https://github.com/unicef/iogt/pull/110/files#r655496249, and make some progress as there was indeed a module called workbox-window that allowed me to 1. Load the service worker as soon as the user comes via a JS enabled browser and 2. communicate with the service worker via messages. The snag however is that the precacheAndRoute feature can only be enabled inside the install event of the serviceworker and not by listening to the messages. Ive tried many workaround for this using the documentation, but it appears that the only way to have this "cache on clicking the download button" feature is how we have done in the current implementation. We will need to register the serviceworker on the click of a button, which then begins the precaching via precacheAndRoute